File: Data Structures & Algorithms/depth-of-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    public int maxDepth(TreeNode root) {
        if (root == null){
            return 0;
        }
    
        int left = maxDepth(root.left);
        int leftDepth = left + 1;

        int right = maxDepth(root.right);
        int rightDepth = right + 1;
        
        return Math.max(leftDepth, rightDepth);
    }
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        self.da_bool = True

        def check(root):
            if not root:
                return 0
            if self.da_bool == False:
                return 0

            left = check(root.left)
            right = check(root.right)

            if abs(left - right) > 1:
                self.da_bool = False
            return 1 + max(left, right)
        
        check(root)
        
        return self.da_bool
